casanovo/denovo/model_runner.py

```diff
@@ -278,22 +265,25 @@ def train(
     tmp_dir.cleanup()
 
 
-def _get_peak_filenames(peak_dir: str) -> List[str]:
+def _get_peak_filenames(path: str) -> List[str]:
     """
-    Get the peak file names in the given directory.
+    Get all matching peak file names from the path pattern.
+
+    Performs cross-platform path expansion akin to the Unix shell (glob, expand
+    user, expand vars).
 
     Parameters
     ----------
-    peak_dir : str
-        The directory in which to find peak files.
+    path : str
+        The path pattern.
 
     Returns
     -------
     List[str]
-        The peak file names from the given directory.
+        The peak file names matching the path pattern.
     """
     return [
-        os.path.join(peak_dir, f)
-        for f in os.listdir(peak_dir)
-        if f.lower().endswith(".mgf")
+        fn
+        for fn in click.utils._expand_args([path])
+        if os.path.splitext(fn.lower())[1] == ".mgf"
```

Comment:
**comment:** I think we should avoid using a private function in click to do the work here.

Maybe we should add our own utility function here? I could be swayed either way.

Reply:
I mainly wanted to get the glob functionality, but after reading some [discussions on click's shell functionality](https://github.com/pallets/click/issues/1096) I found this function that gives user and variable expansion for free compared to using glob.

But it makes sense that we can't depend on having a consistent API for private functions. I'll implement something similar explicitly.